lib/whenever/yaml.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require "yaml"
+
+module Whenever
+  module Output
+    # Generates cron config in an extended YAML format:
+    # ---
+    # version: 1
+    # crons:
+    #   - frequency: "1,2 * * * *"
+    #     name: Jobs::SomeJob
+    #     timezone: UTC
+    #     command: bin/rails runner 'JobRunner::Runner.new(Jobs::SomeJob).run'
+    #     team: core-infrastructure
+    #     retryable: true
+    class Yaml
+      # TODO: validate timezone
+      def self.output_job(time, job)
+        {
```

Comment:
Do we think something like this PR could even be upstreamed? Half of me says no, as it's not crontab and that's the focus of the gem, but the other half says yes, as generating cron in environments like kubernetes is a massive new use case and crontab isn't ideal for that.

If we think this could be upstreamed and we don't want to maintain a fork, this could be made more generic to include any options passed.

Reply:
I think this could potentially be upstreamed, but for it to be nice you'd have to do a bit of refactoring of whenever. I think that could be a nice followup but I don't think it's worth investing in right now, when we're primarily after a tool to generate a format for us.

If this works out, then making it more generic and generally useful would definitely be a good next step!